processor/awscontainerinsightprocessor/processor_test.go

```diff
@@ -0,0 +1,140 @@
+// Copyright  OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package awscontainerinsightprocessor
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"sync"
+	"testing"
+	"time"
+
+	"go.opentelemetry.io/collector/consumer/pdata"
+)
+
+type MockStore struct {
+	RefreshCount int
+	wg           sync.WaitGroup
+}
+
+func (m *MockStore) Decorate(metric pdata.ResourceMetrics, kubernetesBlob map[string]interface{}) bool {
+	return true
+}
+
+func (m *MockStore) RefreshTick(ctx context.Context) {
+	m.RefreshCount++
+	if m.RefreshCount <= 2 {
+		m.wg.Done()
+	}
+}
+
+func (m *MockStore) currentRefreshCount() int {
+	m.wg.Wait()
+	return m.RefreshCount
+}
+
+type LongRunningMockStore struct {
+	RefreshCount int
+}
+
+func (m *LongRunningMockStore) Decorate(metric pdata.ResourceMetrics, kubernetesBlob map[string]interface{}) bool {
+	return true
+}
+
+func (m *LongRunningMockStore) RefreshTick(ctx context.Context) {
+	m.RefreshCount++
+
+	if m.RefreshCount == 2 {
+		wait := make(chan bool)
+		//to simulate a long running store
+		go func() {
+			fmt.Println("before the long running refresh")
+			time.Sleep(20 * time.Second)
+			close(wait)
+			fmt.Println("after the long running refresh")
+		}()
```

Comment:
This may lead to a flaky test on slow/single-threaded CI runners.  It may be better to simply have this wait on a channel that will never receive anything until the context is timed-out.

Should tests be using `fmt.Println()`?  Shouldn't there be an actual assertion that can be made rather than looking for output manually?